content/flow-collections/exercises/_index.md

```diff
@@ -0,0 +1,204 @@
+1. Create a list with at least 10 integers and call your method on the list.
+
+1. Write a `static` method to print out each word in a list that has exactly 5 letters.
+
+   {{% expand "Check your solution" %}}
+   ```csharp
+   static void printFiveLetterWords(List<string> wordlist)
+   {
+      foreach (string word in wordlist)
+      {
+         if (word.Length == 5)
+         {
+            Console.WriteLine(word);
+         }
+      }
+   }
+   ``` 
+   {{% /expand %}}
+
+1. Modify your code to prompt the user to enter the word length for the search.
+
+   {{% expand "Check your solution" %}}
+   ```csharp
+      Console.WriteLine("Enter a word length: ");
+      string numInput = Console.ReadLine();
+      int numChars = int.Parse(numInput);
+
+      // Call the method to print out list words of the chosen length:
+      printXLetterWords(wordList, numChars);
+
+
+      static void printXLetterWords(List<string> wordlist, int length)
+      {
+         foreach (string word in wordlist)
+         {
+            if (word.Length == length)
+            {
+               Console.WriteLine(word);
+            }
+         }
+      }
+   ```
+   {{% /expand %}}
+
+## Dictionary Practice
+Using the `Dictionaries` project in the Exercises repo complete the following:
+
+1. Make a program similar to `GradebookDictionary` that does the following:
+
+   1. It takes in student names and ID numbers (as integers) instead of names and grades.
+   {{% expand "Check your solution" %}}
+   ```csharp
+      Console.WriteLine("Enter your students' names and ID numbers (or ENTER to finish):");
+
+      Console.WriteLine("Student Name: ");
+      newStudent = Console.ReadLine();
+
+      if (newStudent!= "")
+      {
+         Console.WriteLine("ID: ");
+         int newID = int.Parse(Console.ReadLine());
+         students.Add(newID, newStudent);
+      }
+   ```
+   {{% /expand %}}
+   
+
+   1. It takes in student names and ID numbers (as integers) instead of names and grades.
+   {{% expand "Check your solution" %}}
+   ```csharp
+      Console.WriteLine("Enter your students' names and ID numbers (or ENTER to finish):");
+
+      Console.WriteLine("Student Name: ");
+      newStudent = Console.ReadLine();
+
+      if (newStudent!= "")
+      {
+         Console.WriteLine("ID: ");
+         int newID = int.Parse(Console.ReadLine());
+         students.Add(newID, newStudent);
+      }
+   ```
+   {{% /expand %}}
+
+   1. The keys should be the IDs and the values should be the names.
+   {{% expand "Check your solution" %}}
+   ```csharp
+      Console.WriteLine("\nClass roster:");
+
+      foreach (KeyValuePair<int, string> student in students)
+      {
+         Console.WriteLine(student.Value + "'s ID: " + student.Key);
+      }
+
+      Console.WriteLine("Number of students in roster: " + students.Count);
+   ```
+   Review the `Array` and `List` Gradebooks to see how they used loops
+   {{% /expand %}}
+
+   1. Modify the roster printing code accordingly.
```

Comment:
```suggestion
Using the `Dictionaries` project in the Exercises repo make a program similar to `GradebookDictionary` that does the following:

1. It takes in student names and ID numbers (as integers) instead of names and grades.
   {{% expand "Check your solution" %}}
   ```csharp
      Console.WriteLine("Enter your students' names and ID numbers (or ENTER to finish):");

      Console.WriteLine("Student Name: ");
      newStudent = Console.ReadLine();

      if (newStudent!= "")
      {
         Console.WriteLine("ID: ");
         int newID = int.Parse(Console.ReadLine());
         students.Add(newID, newStudent);
      }
   ```
   {{% /expand %}}
   

1. It takes in student names and ID numbers (as integers) instead of names and grades.
   {{% expand "Check your solution" %}}
   ```csharp
      Console.WriteLine("Enter your students' names and ID numbers (or ENTER to finish):");

      Console.WriteLine("Student Name: ");
      newStudent = Console.ReadLine();

      if (newStudent!= "")
      {
         Console.WriteLine("ID: ");
         int newID = int.Parse(Console.ReadLine());
         students.Add(newID, newStudent);
      }
   ```
   {{% /expand %}}

1. The keys should be the IDs and the values should be the names.
   {{% expand "Check your solution" %}}
   ```csharp
      Console.WriteLine("\nClass roster:");

      foreach (KeyValuePair<int, string> student in students)
      {
         Console.WriteLine(student.Value + "'s ID: " + student.Key);
      }

      Console.WriteLine("Number of students in roster: " + students.Count);
   ```
   Review the `Array` and `List` Gradebooks to see how they used loops
   {{% /expand %}}

1. Modify the roster printing code accordingly.
```

Reply:
Didn't need the second nested ordered list.

Reply:
gotcha. done.